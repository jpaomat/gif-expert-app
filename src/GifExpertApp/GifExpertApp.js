import React, { useState } from 'react'
import { AddCategory } from '../components/AddCategory/AddCategory';
import { GifGrid } from '../components/GifGrid/GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {
    const [categories, setCategories] = useState(defaultCategories);
    return (
        <>
            <h2>Gif Expert App</h2>
            <AddCategory setCategories={setCategories} />
            <hr></hr>
            <ol>
                {
                    categories.map(category => (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>
        </>
    )
}




