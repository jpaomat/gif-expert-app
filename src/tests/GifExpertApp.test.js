import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp/GifExpertApp";

describe('Test to GifExpertApp component', () => {
    let wrapper = shallow(<GifExpertApp />);
    test('should create component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should validate if call AddCategory and GifGrid components', () => {
        const props = {
            defaultCategories: [
                'cars',
                'cats'
            ]
        }
        wrapper = shallow(<GifExpertApp {...props} />)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('AddCategory').exists()).toBe(true);
        expect(wrapper.find('GifGrid').length).toBe(props.defaultCategories.length);
    })

})
