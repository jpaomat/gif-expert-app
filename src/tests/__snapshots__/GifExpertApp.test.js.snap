// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to GifExpertApp component should create component 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test to GifExpertApp component should validate if call AddCategory and GifGrid components 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="cars"
      key="cars"
    />
    <GifGrid
      category="cats"
      key="cats"
    />
  </ol>
</Fragment>
`;
